Reported by Guus Sliepen:  "make doc" failed due to missing doctype,
unknown elements, and duplicate element IDs in libraw1394.sgml.

The fix is to declare a recent DTD (matching the one which is used
in current Linux kernel documentation docbooks) and to make the
conflicting element IDs unique.

The latter part of the fix is just temporary.  In order to avoid the
conflict when the documentation is updated the next time, also fix the
kerneldoc comments of the respective API elements:  These are typedefs,
hence kernel-doc needs their comments prepended by "typedef ".

Tested with Gentoo's docbook-xml-dtd 4.5, docbook-xsl-stylesheets
1.75.2, docbook-sgml-utils 0.6.14, and openjade 1.3.2-r1.

Signed-off-by: Stefan Richter <stefanr@s5r6.in-berlin.de>
---

Guus, does this work for you?

 doc/libraw1394.sgml |   10 +++++-----
 src/raw1394.h       |   18 +++++++-----------
 2 files changed, 12 insertions(+), 16 deletions(-)

diff --git a/doc/libraw1394.sgml b/doc/libraw1394.sgml
index d5c11bf..88e97e9 100644
--- a/doc/libraw1394.sgml
+++ b/doc/libraw1394.sgml
@@ -1,6 +1,6 @@
-<!--
-<!DOCTYPE book PUBLIC "-//OASIS//DTD DocBook V3.1//EN">
--->
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE book PUBLIC "-//OASIS//DTD DocBook XML V4.1.2//EN"
+	"http://www.oasis-open.org/docbook/xml/4.1.2/docbookx.dtd" []>
 <book>
   <bookinfo>
 
@@ -3184,7 +3184,7 @@
 </refsect1>
 </refentry>
 
-<refentry id="API-int">
+<refentry id="API-req_callback_t">
 <refentryinfo>
  <title>LINUX</title>
  <productname>Kernel Hackers Manual</productname>
@@ -3231,7 +3231,7 @@
 </refsect1>
 </refentry>
 
-<refentry id="API-int">
+<refentry id="API-arm_req_callback_t">
 <refentryinfo>
  <title>LINUX</title>
  <productname>Kernel Hackers Manual</productname>
diff --git a/src/raw1394.h b/src/raw1394.h
index 7bfeb05..99d8d40 100644
--- a/src/raw1394.h
+++ b/src/raw1394.h
@@ -699,12 +699,11 @@ typedef int (*fcp_handler_t)(raw1394handle_t, nodeid_t nodeid, int response,
 fcp_handler_t raw1394_set_fcp_handler(raw1394handle_t handle, fcp_handler_t new_h);
 
 /**
- * req_callback_t - This is the general request handler
- * @req_callback_t: This is the general request handler
+ * typedef req_callback_t - This is the general request handler
  *
- * It is used by the default tag handler
- * when a request completes, it calls the callback and passes it the data
- * pointer and the error code of the request.
+ * Used by the default tag handler.  When a request completes, it calls
+ * the callback and passes it the libraw1394 handle, the data pointer,
+ * and the error code of the request.
  **/
 typedef int (*req_callback_t)(raw1394handle_t, void *data,
                               raw1394_errcode_t err);
@@ -715,13 +714,10 @@ struct raw1394_reqhandle {
 };
 
 /**
- * arm_req_callback_t - This is the general arm-request handle
- * @arm_req_callback_t: This is the general arm-request handle
- * @handle: libraw1394 handle
+ * typedef arm_req_callback_t - This is the general arm-request handle
  *
- * (arm = address range mapping)
- * It is used by the default arm-tag handler when a request has been 
- * received, it calls the arm_callback.
+ * Used by the default arm-tag handler (arm = address range mapping).
+ * Called when a request has been received.
  **/
 typedef int (*arm_req_callback_t) (raw1394handle_t,
                                    struct raw1394_arm_request_response *arm_req_resp,
-- 
1.6.4.4


-- 
Stefan Richter
-=====-==-=- ---= -=--=
http://arcgraph.de/sr/

